custom_components/tuya_local/devices/plantsio_ivy_smart_planter.yaml

```diff
@@ -0,0 +1,431 @@
+# Datapoints documented at:
+# https://gist.github.com/thewade/ef9f6014f13932bd2e77d43331e2027d
+
+name: Smart Planter
+products:
+  - id: 8n3q0y4cwov8ifxt
+    name: PlantsIO Ivy Smart Planter
+primary_entity:
+  entity: sensor
+  name: Water level
+  dps:
+    - id: 108
+      type: integer
+      name: sensor
+      unit: mL
+      class: measurement
+      mapping:
+        - dps_val: 0
+          icon: "mdi:cup-outline"
+        - icon: "mdi:cup"
+    - id: 107
+      type: string
+      name: version
+secondary_entities:
+  # Sensor entities
+  - entity: sensor
+    name: Ambient light
```

Comment:
Suggest to remove this name, so that pre-defined translations can be used.

Reply:
When I do that all my sensors have no names:

![image](https://github.com/make-all/tuya-local/assets/744796/2cb5f4ad-5fb9-4a38-a8c7-bac438479897)

And I end up getting entity IDs like:

sensor.smart_planter
sensor.smart_planter_2
sensor.smart_planter_3

Am I doing something wrong?

This is my current code for the illuminance sensor:

```
  - entity: sensor
    class: illuminance
    dps:
      - id: 103
        type: integer
        name: sensor
        unit: lx
        class: measurement
```

Reply:
This is maybe a recently introduced bug due to a side effect of the fix for #1856.  It may be better for me to revert that change.
Unfortunately this was difficult to detect, as existing entity names do not change once they are setup, only newly added ones are affected.